### PR TITLE
Remove PortablePdbReader except trace cleanup

### PR DESCRIPTION
**Pull request by RozzaysRed**
Summary of the changes (Less than 80 chars)
 - Removes PortablePdbReader except the Dispose method so it should clean up the stack trace

Addresses #19441

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
Thanks for the PR, @RozzaysRed.
@pranavkm can you please look into this? Thanks!

**Comment by mkArtakMSFT** [comment]
Actually, the related issue is `area-middleware` so feel free to update this too, @pranavkm.

**Comment by davidfowl** [comment]
Does this affect the error page in any way?

**Comment by pranavkm** [comment]
@davidfowl likely not - we added this type because .NET Fx was unable to natively read portable pdbs. https://github.com/dotnet/aspnetcore/blob/master/src/Mvc/test/Mvc.FunctionalTests/ErrorPageTests.cs has some tests that should cover end-to-end scenarios to an extent.

File: src/Shared/StackTrace/StackFrame/PortablePdbReader.cs [hunk 1]
```diff
@@ -17,111 +17,6 @@ internal class PortablePdbReader : IDisposable
...
-            using (var pdbStream = File.OpenRead(pdbPath))
-            {
-                return pdbStream.ReadByte() == 'B' &&
-                    pdbStream.ReadByte() == 'S' &&
-                    pdbStream.ReadByte() == 'J' &&
-                    pdbStream.ReadByte() == 'B';
-            }
-        }
-
         public void Dispose()
```
Comment (pranavkm) [inline src/Shared/StackTrace/StackFrame/PortablePdbReader.cs]: Can this type be removed?

**Comment by davidfowl** [comment]
@pranavkm Kewl.

@RozzaysRed can you show a screenshot of the DeveloperExceptionPage after this change?

File: src/Shared/StackTrace/StackFrame/PortablePdbReader.cs (hunk 1 above)
Comment (RozzaysRed) [reply src/Shared/StackTrace/StackFrame/PortablePdbReader.cs]: I went ahead and just removed the PortablePdbReader.cs

**Comment by RozzaysRed** [comment]
@davidfowl sure as soon I figure out how to get this page, I will take the requested screenshot.  I'm assuming its a dummy project and enabling that middleware?  Pardon me for asking but just want to make sure I get you exactly what you are looking for.

**Comment by pranavkm** [comment]
@RozzaysRed you can run the https://github.com/dotnet/aspnetcore/blob/master/src/Mvc/samples/MvcSandbox and include screenshots for the result when an exception is thrown from 

* a controller
* a Razor Page \ view 

The sample is already set up to include the developer error page middleware

**Comment by RozzaysRed** [comment]
image

**Comment by pranavkm** [comment]
Would you mind including screenshot of an exception thrown from inside a view? We've had issues with it in the past, so it would be good to confirm that looks correct.

File: src/Shared/test/Shared.Tests/Microsoft.AspNetCore.Shared.Tests.csproj [hunk 2]
```diff
@@ -8,29 +8,33 @@
...
-    <Compile Include="$(SharedSourceRoot)ParameterDefaultValue\*.cs" Link="Shared\ParameterDefaultValue\%(Filename)%(Extension)"/>
-    <Compile Include="$(SharedSourceRoot)PropertyActivator\*.cs" Link="Shared\PropertyActivator\%(Filename)%(Extension)"/>
-    <Compile Include="$(SharedSourceRoot)PropertyHelper\*.cs" Link="Shared\PropertyHelper\%(Filename)%(Extension)"/>
-    <Compile Include="$(SharedSourceRoot)SecurityHelper\**\*.cs" Link="Shared\SecurityHelper\%(Filename)%(Extension)"/>
-    <Compile Include="$(SharedSourceRoot)StackTrace\StackFrame\**\*.cs" Link="Shared\StackTrace\%(Filename)%(Extension)"/>
-    <Compile Include="$(SharedSourceRoot)TypeNameHelper\StackFrame\**\*.cs" Link="Shared\TypeNameHelper\%(Filename)%(Extension)"/>
-    <Compile Include="$(SharedSourceRoot)ValueStopwatch\**\*.cs" Link="Shared\ValueStopwatch\%(Filename)%(Extension)"/>
-    <Compile Include="$(SharedSourceRoot)WebEncoders\**\*.cs" Link="Shared\WebEncoders\%(Filename)%(Extension)"/>
+    <Compile Include="$(SharedSourceRoot)ActivatorUtilities\**\*.cs" Link="Shared\ActivatorUtilities\%(Filename)%(Extension)" />
```
Comment (pranavkm) [inline src/Shared/test/Shared.Tests/Microsoft.AspNetCore.Shared.Tests.csproj]: Could you undo changes to this file? I don't think any of these are necessary.

**Review by pranavkm — COMMENTED** [review]
Looks great!

**Comment by RozzaysRed** [comment]
@pranavkm sorry about that, I thought I had pasted the file that had both.  Here is the view: 
image

**Comment by pranavkm** [comment]
Thanks!

File: src/Shared/test/Shared.Tests/Microsoft.AspNetCore.Shared.Tests.csproj (hunk 2 above)
Comment (RozzaysRed) [reply src/Shared/test/Shared.Tests/Microsoft.AspNetCore.Shared.Tests.csproj]: Changes should be reverted back for this file.

**Review by pranavkm — APPROVED** [review]
Thanks!

**Pull request merged**